spec/classes/hyperglass_server_spec.rb

```diff
@@ -15,19 +15,191 @@
 
       context 'with all defaults' do
         it { is_expected.to compile.with_all_deps }
-        it { is_expected.to contain_class('Hyperglass::Server::Config') }
-        it { is_expected.to contain_class('Hyperglass::Server::Dependencies') }
-        it { is_expected.to contain_class('Hyperglass::Server::Install') }
-        it { is_expected.to contain_class('Hyperglass::Server::Service') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/commands.yaml').with_ensure('file').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/devices.yaml').with_ensure('file').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/hyperglass.yaml').with_ensure('file').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/static/custom').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/static/images/favicons').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/static/images').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/static/ui').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass/static').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
-        it { is_expected.to contain_file('/opt/hyperglass/hyperglass-server/hyperglass').with_ensure('directory').with_owner('hyperglass').with_group('hyperglass') }
+
+        it { is_expected.to contain_class('hyperglass::agent') }
+
+        it { is_expected.to contain_class('nginx') }
+
+        it { is_expected.to contain_class('python').with_version('3').with_dev('present') }
+
+        it { is_expected.to contain_package('gcc').with_ensure('installed') }
+
+        it { is_expected.to contain_class('redis::globals').with_scl('rh-redis5').with_before('Class[Redis]') }
+
+        it { is_expected.to contain_class('redis').with_manage_repo(true) }
+
+        it {
+          is_expected.to contain_yumrepo('yarn').with(
+            {
+              'ensure'   => 'present',
+              'baseurl'  => 'https://dl.yarnpkg.com/rpm/',
+              'gpgcheck' => 1,
+              'gpgkey'   => 'https://dl.yarnpkg.com/rpm/pubkey.gpg',
+              'descr'    => 'Yarn Repository',
+            }
+          )
+        }
+
+        it { is_expected.to contain_package('yarn').with_ensure('present').with_require('Yumrepo[yarn]') }
```

Comment:
`.that_requires` instead of `.with_requires`

Reply:
eh? with works and it is more natural to read since we are translating `.with_param` it is odd to have some `.with_param` and some `.that_metaparam`.

Reply:
There is a difference. `that_requires` checks the actual compiled catalog where `with_require` only checks the parameter. AFAIK it also checks if that resource is actually found in the catalog.

Reply:
Both check the compiled catalog. Seems that neither are the way to go though and we should use the relationship matches that @alexjfisher mentioned above.

Reply:
search and replace done